From 14d8fd5e9f3eddd6aca2e7b9bc236fa74054a46e Mon Sep 17 00:00:00 2001
From: "kidd-kw.chen" <kidd-kw.chen@mediatek.com>
Date: Fri, 14 Jan 2022 16:25:11 +0800
Subject: [PATCH] nnapi: add dummy impl of validateRequestfor

	workaround for build nnapi-cts when using armnn (android-nn-driver),
	add dummy impl of
	validateRequestfor<const V1_0::Request&, const V1_3::Model, bool>()

	Signed-off-by: kidd <kidd-kw.chen@mediatek.com>

%% original patch: 0001-nnapi-add-dummy-impl-bool-validateRequest-V1_0-Reque.patch
---
 nn/common/ValidateHal.cpp | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/nn/common/ValidateHal.cpp b/nn/common/ValidateHal.cpp
index 6470fbce1..783562cb6 100644
--- a/nn/common/ValidateHal.cpp
+++ b/nn/common/ValidateHal.cpp
@@ -859,6 +859,23 @@ bool validateRequest(const V1_3::Request& request, const V1_3::Model& model,
             validatePools(request.pools, HalVersion::V1_3));
 }
 
+template <>
+bool validateRequest(const V1_0::Request& request, const V1_3::Model& model,
+                     bool allowUnspecifiedOutput) {
+	MemoryAccessVerifier poolVerifier(request.pools);
+    LOG(ERROR) << "Unexpected operation\n";
+    return (validateRequestArguments(request.inputs, model.main.inputIndexes, model.main.operands,
+                                     request.pools,
+                                     //allowUnspecified=
+									 false, "input") &&
+            validateRequestArguments(
+					request.outputs, model.main.outputIndexes, model.main.operands,
+					poolVerifier, 
+					allowUnspecifiedOutput, 
+					"output") &&
+            validatePools(request.pools, HalVersion::V1_3));
+}
+
 bool validateMemoryDesc(const V1_3::BufferDesc& desc,
                         const hidl_vec<sp<V1_3::IPreparedModel>>& preparedModels,
                         const hidl_vec<V1_3::BufferRole>& inputRoles,
-- 
2.18.0

